fbw-a380x/src/wasm/systems/a380_systems/src/fuel/fuel_quantity_management_system.rs

```diff
@@ -237,10 +307,97 @@ impl RefuelApplication {
         }
     }
 
+    fn lookup_trim_fuel_from_target_fuel_range(
+        &mut self,
+        category_index: usize,
+        total_fuel_desired_rounded: u32,
+        zero_fuel_weight_cg_rounded: u32,
+    ) -> Mass {
+        let zfw_category = &self.trim_tank_map.trim_tank_targets.tables[category_index];
+        let target_fuel_range = zfw_category.get(&total_fuel_desired_rounded).unwrap();
+
+        if let Some(cg_index) = self
+            .trim_tank_map
+            .params
+            .target_zfw_cg
+            .iter()
+            .position(|&x| x == zero_fuel_weight_cg_rounded)
+        {
+            return Mass::new::<kilogram>(target_fuel_range[cg_index] as f64);
+        } else {
+            return Mass::new::<kilogram>(target_fuel_range[0] as f64);
+        }
```

Comment:
This should be semantically the same:
```suggestion
        let cg_index = self
            .trim_tank_map
            .params
            .target_zfw_cg
            .iter()
            .position(|&x| x == zero_fuel_weight_cg_rounded)
            .unwrap_or_default();
        Mass::new::<kilogram>(target_fuel_range[cg_index] as f64)
```